web/yo/app/scripts/directives/addtherapy.js

```diff
@@ -1,17 +1,27 @@
 'use strict';
 angular.module('oncokbApp')
-    .directive('addTherapy', function ($rootScope, DatabaseConnector, _, $q, FirebaseModel, firebaseConnector, mainUtils, $window) {
+    .directive('addTherapy', function ($rootScope, DatabaseConnector, _, $q, FirebaseModel, firebaseConnector, drugMapUtils, $window) {
         return {
             templateUrl: 'views/addTherapy.html',
             restrict: 'E',
+            scope:{
+                modifyName: '=',
+                tiRef: '=',
+                treatmentRef: '=',
+                saveCallbackInGene: '&addNewTherapy',
```

Comment:
Why do you need to differentiate where the component is used?

Reply:
These two functions are defined in two controllers. One is Gene Controller, the other one is ModifyTherapy controller. I couldn't define both as saveCallback.

Reply:
Back to previous discussion, the component should not aware of where it will be used. You don't need to define `add-new-therapy` property for GeneController and `save-modification` for ModifyController. They should all be `save-callback`.